File: continuum-idl/src/main/java/org/kinotic/continuum/idl/api/decorators/DecoratorTarget.java
package org.kinotic.continuum.idl.api.decorators;

/**
 * Created by Navíd Mitchell 🤪 on 4/23/23.
 */
public enum DecoratorTarget {

    /** Class, interface, or enum declaration */
    TYPE,

    /** Field declaration (includes enum constants) */
    FIELD,

    /** Function declaration */
    FUNCTION,

    /** Formal parameter declaration */
    PARAMETER,

    /** Constructor declaration */
    CONSTRUCTOR,

    /** Namespace declaration */
    NAMESPACE

}

File: continuum-clients/continuum-idl-js/src/api/decorators/DecoratorTarget.ts


export enum DecoratorTarget {
    /** Class, interface, or enum declaration */
    TYPE = "TYPE",
    /** Field declaration (includes enum constants) */
    FIELD = "FIELD",
    /** Function declaration */
    FUNCTION = "FUNCTION",
    /** Formal parameter declaration */
    PARAMETER = "PARAMETER",
    /** Constructor declaration */
    CONSTRUCTOR = "CONSTRUCTOR",
    /** Namespace declaration */
    NAMESPACE = "NAMESPACE"
}